scripts/geoid_tool.py

```
@@ -0,0 +1,254 @@
+                        # saves out to log file and outputs to command line.
+                        handlers=[logging.FileHandler(join(log_dir, f'{log_prefix}-r{vnum}.log')),
+                                  logging.StreamHandler(sys.stdout)]
+                        )
+    
+    log = logging.getLogger(__name__)
+
+    if debug:
+        log.setLevel(logging.DEBUG)  
+
+    # check for white spaces
+    if len([i for i in glob(join(in_dir, '*')) if ' ' in i]) > 0:
+        log.warning('White spaces found in file paths. Try and remove them?')
+        replace_white_spaces(in_dir)
+
+    # Check what kind of file was given by user
+    user_file_type = None
+    pc_ext = {'.las', '.laz'}
+    raster_ext = {'.tif', '.tiff'}
+    extension = in_dir.lower().split('.')[-1]
+    if f".{extension}" in pc_ext:
+        user_file_type = 'Point Cloud'
+    elif f".{extension}" in raster_ext:
+        user_file_type = 'Raster'
+    else:
+        user_file_type = None
+        raise Exception(f'User data needs to be in following format:{pc_ext} or {raster_ext}')
+    
+    # Make sure correct transformation selected, is their lidar in geoid, and they used the "to_ellipsoid" command mistakenly
+    epsg_object = CRS.from_epsg(epsg)
+    datum = epsg_object.datum.name
+    datum = code_datum(datum)
+    if 'WGS'==datum in datum and transform_command == 'to_geoid':
+        raise Exception(f'Your target is to get to ellipsoid based on input. You should use "to_ellipsoid".')
+    elif 'NAD' ==datum and transform_command == 'to_ellipsoid':
+        raise Exception(f'Your target is to get to geoid based on input. You should use "to_geoid".')
+    else:
+        pass
+
+    # run main functions
+    # Either Point Cloud or Raster...
+    if user_file_type == 'Point Cloud':
+        log.info(f'Starting run with {user_file_type} data type using PDAL.')
+
+        # Do not allow transformation if they are already the same datum
+        with laspy.open(in_dir) as las:
```

This goes for both the PC and the raster method, but perhaps in the unlikely case user data does not have a vertical datum it should throw a helpful error message here.